test/Autofac.Test/Features/Decorators/DecoratorMiddlewareTests.cs

```diff
@@ -14,19 +14,19 @@ public class DecoratorMiddlewareTests
     public void ResolveOperationDoesNotImplementIDependencyTrackingResolveOperation_DecoratorMiddlewareStoppedEarly()
     {
         var decoratorService = new DecoratorService(typeof(object));
-        var middleware = new DecoratorMiddleware(decoratorService, Mock.Of<IComponentRegistration>());
-        var contextMock = new Mock<ResolveRequestContext>();
-        var registrationMock = new Mock<IComponentRegistration>();
-        contextMock.Setup(context => context.Instance).Returns(new object());
-        contextMock.Setup(context => context.Registration).Returns(registrationMock.Object);
-        registrationMock.Setup(registration => registration.Options).Returns(RegistrationOptions.None);
+        var middleware = new DecoratorMiddleware(decoratorService, Substitute.For<IComponentRegistration>());
+        var contextMock = Substitute.For<ResolveRequestContext>();
+        var registrationMock = Substitute.For<IComponentRegistration>();
+        contextMock.Instance.Returns(new object());
+        contextMock.Registration.Returns(registrationMock);
+        registrationMock.Options.Returns(RegistrationOptions.None);
 
-        middleware.Execute(contextMock.Object, context => { });
+        middleware.Execute(contextMock, context => { });
 
-        contextMock.Verify(context => context.Instance, Times.Once);
-        contextMock.Verify(context => context.Registration.Options, Times.Once);
+        _ = contextMock.Received(1).Instance;
```

Comment:
This discard construct reads kind of weird. Is there an error of some nature if the discard part is omitted?

Reply:
Good spot! My mistake.